Userland/Applications/ImageViewer/ViewWidget.cpp

```diff
@@ -77,33 +93,28 @@ void ViewWidget::navigate(Directions direction)
         }
     }
 
-    auto current_index = m_files_in_same_dir.find_first_index(m_path);
-    if (!current_index.has_value()) {
+    current_index = m_files_in_same_dir.find_first_index(m_path);
+    return current_index;
+}
+
+void ViewWidget::navigate(Directions direction)
+{
+    auto present_index = get_current_index();
+    if (!present_index.has_value()) {
         return;
     }
 
-    size_t index = current_index.value();
     if (direction == Directions::Back) {
-        if (index == 0) {
-            GUI::MessageBox::show(window(), "This is the first file.", "Cannot open image", GUI::MessageBox::Type::Error);
-            return;
-        }
-
-        index--;
+        current_index.value()--;
     } else if (direction == Directions::Forward) {
-        if (index == m_files_in_same_dir.size() - 1) {
-            GUI::MessageBox::show(window(), "This is the last file.", "Cannot open image", GUI::MessageBox::Type::Error);
-            return;
-        }
-
-        index++;
+        current_index.value()++;
```

Comment:
This is an odd way to do it, the check at the top (seems to) be ensuring that this has a value, so please:
- copy it out into a local variable at the top
- modify said local variable
- write back into `m_current_index` at the end of this function

Otherwise this would be very confusing.